spec/lib/services/messenger-spec.js

```diff
@@ -4,10 +4,81 @@
 'use strict';
 
 describe('Messenger', function () {
-    var subscription, Errors, ErrorEvent, IpAddress, Constants;
-
-    helper.before();
-
+    var Errors, ErrorEvent, IpAddress, Constants,
+        testData = { hello: 'world' },
+        sandbox = sinon.sandbox.create();
+    
+    helper.before(function(context) {
+        function mockPromise () {
+            this.addCallback = sandbox.spy(function(callback) {
+                callback({});
+            });
+        }
+        
+        var exchange = {
+            publish: sandbox.stub().resolves()
+        };
+        
+        var queue = {
+            subscribe: sandbox.spy(function(a,callback) {
+                callback(testData,{},{});
+                return new mockPromise();
+            }),
+            bind: sandbox.spy(function(a,b,callback) {
+                callback();
+            })
+        };
+        
+        context.statsd = {
+            stop: sandbox.stub().resolves(),
+            start: sandbox.stub().resolves(),
+            sanitize: sandbox.stub().resolves(),
+            timing: sandbox.stub().resolves()
+        };
+        
+        context.Subscription = function (a,b) {
+            this.dispose = sandbox.stub();
+        }
+        
+        context.Connection = function (options, clientOptions, label) {
```

Comment:
'label' is defined but never used.<br>'clientOptions' is defined but never used.<br>'options' is defined but never used.